src/Application/src/CalculatorApplication.cpp

```diff
@@ -6,64 +6,125 @@
  */
 ////////////////////////////////////////////////////////////////////////////
 
+#include <future>
 #include <iostream>
 #include <limits>
 #include <sstream>
 #include <string>
-
+#include <thread>
 
 #include "CalculatorApplication.h"
 #include "CalculatorApplicationFactory.h"
 #include "CalculatorStrings.h"
-namespace calculator
+
+namespace calculator {
+
+// *****************************************************************************/
+// ***************** CalculatorApplication public methods **********************/
+// *****************************************************************************/
+
+int CalculatorApplication::calculate(float firstNumber, float secondNumber, char operation)
 {
-    int CalculatorApplication::calculate(float a, float b, char op)
+    int result = 0;
+
+    if(limitCheck(firstNumber) && limitCheck(secondNumber))
     {
         calculator::CalculatorApplicationFactory calculatorAppFactory;  
-
-        if(limitCheck(a) && limitCheck(b))
+        auto calculator = calculatorAppFactory.createCalculator(firstNumber, secondNumber, operation); //variables placed into createCalc function and placed into calculator variable
+        if(calculator)
         {
-            auto calculator = calculatorAppFactory.createCalculator(a, b, op); //variables placed into createCalc function and placed into calculator variable
-            if(calculator)
-            {
-                std::cout << calculator->toString() << std::endl; //if calculator returned from function is valid, point to string function and output expression and result
-            }
-            else
-            {
-                std::cout << CalculatorStrings::ERROR_MESSAGE_INVALID_INPUT << std::endl;
-            }
+            result = calculator->getResult();
+            std::cout << calculator->toString() << std::endl; //if calculator returned from function is valid, point to string function and output expression and result
         }
         else
         {
-            return -1;
+            std::cout << CalculatorStrings::ERROR_MESSAGE_INVALID_INPUT << std::endl;
         }
-        return 0;
     }
-
-    std::string CalculatorApplication::toString()
+    else
     {
-        return std::to_string(firstNumber_) + " " + op_ + " " + std::to_string(secondNumber_) + " = " + std::to_string(getResult());
+        result = -1;
     }
 
-    bool CalculatorApplication::limitCheck(float a)
-    {
-        if(a > std::numeric_limits<float>::max())
-        {
-            std::cerr << CalculatorStrings::ERROR_MESSAGE_INPUT_TOO_LARGE << std::endl;
-            return false;
-        }
-        else if(a < std::numeric_limits<float>::min())
+    return result;
+}
+
+void CalculatorApplication::runCalculator()
+{
+    Parser parser;
+        float answer;
+        Expression parsedExpression;
+
+        auto tempVector = std::async(&Parser::createVector, parser, parser.getUserInput());
```

Comment:
Avoid using vector in variable names. It is not very descriptive. Prefer to use the plural of the class. I give a longer spiel in Jared's PR that you can read. 

E.g., use `tempExpressionUnits` instead of `tempExpressionUnitVector` and don't just do `tempVector` because then I think "A vector of what? What does that list contain?" It makes it hard to follow later because I don't understand what the variable actually is.